src/requirement-types.d.ts

```diff
@@ -165,3 +165,10 @@ type GroupedRequirementFulfillmentReport = {
   readonly specific: string;
   readonly reqs: readonly RequirementFulfillment[];
 };
+
+type GroupedRequirementFulfillmentReportWithProgress = GroupedRequirementFulfillmentReport & {
+  readonly dangerouslyFulfilled: number;
+  readonly totalRequired: number;
+  readonly safeProgress: number;
+  readonly dangerousProgress: number;
+};
```

Comment:
```suggestion
type GroupedRequirementFulfillmentReportWithProgress = GroupedRequirementFulfillmentReport & Readonly<{
  dangerouslyFulfilled: number;
  totalRequired: number;
  safeProgress: number;
  dangerousProgress: number;
}>;
```

Reply:
you, andrew, and I all have like a 1/3 of the commits on this branch so just commit any changes you want lol